src/macros.jl

```diff
@@ -309,17 +304,24 @@ end
 # end
 
 # three-argument constructconstraint! is used for two-sided constraints.
-function constructconstraint!(aff::AffExpr, lb::Real, ub::Real)
+constructconstraint!(_error::Function, v::Variable, lb::Real, ub::Real) = SingleVariableConstraint(v, MOI.Interval(lb, ub))
+
+function constructconstraint!(_error::Function, aff::AffExpr, lb::Real, ub::Real)
     offset = aff.constant
     aff.constant = 0.0
     AffExprConstraint(aff,MOI.Interval(lb-offset,ub-offset))
 end
 
-# constructconstraint!(aff::Variable, lb::Real, ub::Real) = constructconstraint!(convert(AffExpr,v),lb,ub)
+constructconstraint!(_error::Function, q::QuadExpr, lb, ub) = error("Two-sided quadratic constraints not supported. (Try @NLconstraint instead.)")
 
-constructconstraint!(q::QuadExpr, lb, ub) = error("Two-sided quadratic constraints not supported. (Try @NLconstraint instead.)")
+function constructconstraint!(_error::Function, expr, lb, ub)
+    @assert !(lb isa Number) || !(ub isa Number)
```

Comment:
What's the point of this assert?

Reply:
It could happen that a call is dispatched to this method with `lb` and `ub` that are not `Number` if there is not `constructconstraint!` method defined for the type of `expr`. In that case, no error will be thrown that the `@constraint` will silently do nothing. We want to avoid that